dkg/src/transcript_aggregation.rs

```diff
@@ -0,0 +1,176 @@
+// Copyright © Aptos Foundation
+
+use crate::{types::DKGNodeRequest, DKGMessage};
+use anyhow::ensure;
+use aptos_consensus_types::common::Author;
+use aptos_crypto::{bls12381, Uniform};
+use aptos_infallible::Mutex;
+use aptos_reliable_broadcast::BroadcastStatus;
+use aptos_types::{
+    dkg::{DKGNode, DKGTrait, DKGTranscriptMetadata, DummyDKG, DummyDKGTranscript},
+    epoch_state::EpochState,
+    validator_verifier::{ValidatorConsensusInfo, ValidatorVerifier},
+};
+use move_core_types::account_address::AccountAddress;
+use std::{collections::HashSet, sync::Arc};
+
+pub struct TranscriptAggregator<S: DKGTrait> {
+    pub contributors: HashSet<AccountAddress>,
+    pub trx: Option<S::Transcript>,
+}
+
+impl<S: DKGTrait> Default for TranscriptAggregator<S> {
+    fn default() -> Self {
+        Self {
+            contributors: HashSet::new(),
+            trx: None,
+        }
+    }
+}
+
+pub struct TranscriptAggregationState<DKG: DKGTrait> {
+    trx_aggregator: Mutex<TranscriptAggregator<DKG>>,
+    dkg_pub_params: DKG::PublicParams,
+    epoch_state: Arc<EpochState>,
+}
+
+impl<DKG: DKGTrait> TranscriptAggregationState<DKG> {
+    pub fn new(dkg_pub_params: DKG::PublicParams, epoch_state: Arc<EpochState>) -> Self {
+        //TODO: take threshold as a parameter.
```

Comment:
threshold is part of `dkg_pub_params`?

Reply:
to be clear this threshold refers to the aggregation threshold as you pointed below.
It can be an DKG on-chain config.